src/mrpro/operators/CartesianSamplingOp.py

```diff
@@ -50,23 +50,29 @@ def __init__(self, encoding_matrix: SpatialDimension[int], traj: KTrajectory) ->
             kx_idx = ktraj_tensor[-1, ...].round().to(dtype=torch.int64) + sorted_grid_shape.x // 2
         else:
             sorted_grid_shape.x = ktraj_tensor.shape[-1]
-            kx_idx = repeat(torch.arange(ktraj_tensor.shape[-1]), 'k0->other k2 k1 k0', other=1, k2=1, k1=1)
+            kx_idx = repeat(
+                torch.arange(ktraj_tensor.shape[-1]), 'k0->other coils k2 k1 k0', other=1, coils=1, k2=1, k1=1
+            )
 
         if traj_type_kzyx[-2] == TrajType.ONGRID:  # ky
             ky_idx = ktraj_tensor[-2, ...].round().to(dtype=torch.int64) + sorted_grid_shape.y // 2
         else:
             sorted_grid_shape.y = ktraj_tensor.shape[-2]
-            ky_idx = repeat(torch.arange(ktraj_tensor.shape[-2]), 'k1->other k2 k1 k0', other=1, k2=1, k0=1)
+            ky_idx = repeat(
+                torch.arange(ktraj_tensor.shape[-2]), 'k1->other coils k2 k1 k0', other=1, coils=1, k2=1, k0=1
+            )
 
         if traj_type_kzyx[-3] == TrajType.ONGRID:  # kz
             kz_idx = ktraj_tensor[-3, ...].round().to(dtype=torch.int64) + sorted_grid_shape.z // 2
         else:
             sorted_grid_shape.z = ktraj_tensor.shape[-3]
-            kz_idx = repeat(torch.arange(ktraj_tensor.shape[-3]), 'k2->other k2 k1 k0', other=1, k1=1, k0=1)
+            kz_idx = repeat(
+                torch.arange(ktraj_tensor.shape[-3]), 'k2->other coils k2 k1 k0', other=1, coils=1, k1=1, k0=1
+            )
 
         # 1D indices into a flattened tensor.
         kidx = kz_idx * sorted_grid_shape.y * sorted_grid_shape.x + ky_idx * sorted_grid_shape.x + kx_idx
-        kidx = rearrange(kidx, '... kz ky kx -> ... 1 (kz ky kx)')
+        kidx = rearrange(kidx, '... 1 kz ky kx -> ... 1 (kz ky kx)')
```

Comment:
maybe more obvious by 
```suggestion
        kidx = rearrange(kidx, '... coils kz ky kx -> ... coils (kz ky kx)', coils=1)
```

Reply:
remvoed coils here